connectors/sources/redis.py

```diff
@@ -0,0 +1,283 @@
+            decode_responses=True,
+            max_connections=MAX_POOL_SIZE,
+        )
+        yield redis.Redis(connection_pool=pool)
+
+    async def validate_database(self, db, redis_client):
+        try:
+            await redis_client.execute_command("SELECT", db)
+            return await redis_client.ping()
+        except Exception as exception:
+            self._logger.warning(f"Database {db} not found. Error: {exception}")
+
+    async def get_databases(self):
+        """Returns number of databases from config_get response
+
+        Returns:
+            list: List of databases
+        """
+        if self.database != ["*"]:
+            return set(self.database)
+        databases = []
+        try:
+            async with self.client() as redis_client:
+                database = await redis_client.config_get("databases")
+                databases = list(range(int(database.get("databases", 1))))
+        except Exception as exception:
+            self._logger.warning(
+                f"Something went wrong while fetching database list. Error: {exception}",
+                exc_info=True,
+            )
+        return databases
+
+    async def _get_paginated_keys(self, redis_client, pattern, _type=None):
+        """Make a paginated call for fetching database keys.
+
+        Args:
+            redis_client: redis client object
+            pattern: keyname or pattern
+            _type: Datatype for filter data
+
+        Yields:
+            list: list of keys in a database
+        """
+        cursor, keys = await redis_client.scan(
+            match=pattern, count=PAGE_SIZE, _type=_type
+        )
+        yield keys
+
+        while cursor != 0:
+            cursor, keys = await redis_client.scan(
+                cursor=cursor, match=pattern, count=PAGE_SIZE
+            )
+            yield keys
+
+    async def get_key_value(self, redis_client, key, key_type):
+        """Fetch value of key for database.
+
+        Args:
+            redis_client: redis client object
+            key: Name of key
+            key_type: Type of key
+
+        Yields:
+            Value of key
+        """
+        try:
+            if key_type == KeyType.HASH.value:
+                return await redis_client.hgetall(key)
+            elif key_type == KeyType.STREAM.value:
+                return await redis_client.xread({key: "0"})
+            elif key_type == KeyType.LIST.value:
+                return await redis_client.lrange(key, 0, -1)
+            elif key_type == KeyType.SET.value:
+                return await redis_client.smembers(key)
+            elif key_type == KeyType.ZSET.value:
+                return await redis_client.zrange(key, 0, -1, withscores=True)
+            elif key_type == KeyType.JSON.value:
+                value = await redis_client.execute_command("JSON.GET", key)
+                return json.loads(value)
+            else:
+                value = await redis_client.get(key)
```

Comment:
Can we simply use `get` here? If so, we can use `mget` as well

Reply:
we can use `get` or `mget` only for `string` type of keys. And if we'll switch to `mget` then we need to iterate over all the keys to find the string type and store it in a list to pass. I guess in that case iteration will be increase.